src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs

```diff
@@ -190,6 +189,8 @@ public sealed class PythonAnalyzer : IPythonAnalyzer, IDisposable {
                     }
                 }
 
+                _queueTracker.Add(entry.Module.FilePath);
```

Comment:
The only source of truth of how many modules are left to be analyzed is `walker.Remaining` of the walker that has `_version == walker.Version`. So this is the parameter that must be checked before every `_progress.ReportRemaining()` call, and that should be enough.

Reply:
Basically I track `queued` vs `completed` pairs in the log for the same path and they seem to match.  Let me try your suggestion though.